### ISSUE TITLE
Admin functionality: get all resources for a specific user

### ISSUE THREAD

**Issue by anne17** [spraakbanken/mink-backend#81]
This would be great for debugging/helping a user!
We could add a parameter to `/resource-info` which lets an admin filter out a specific user's resources.

**Comment by arildm** [comment]
Yes, this would be helpful. I agree it's not clear what the best solution is.

A few thoughts:

- What I do currently to find a user's resources: go to SB Auth admin, find the user and list their grants, then look for each `mink-<...>` id in the Mink admin GUI (you could copy-paste the id, but I find it's usually enough to just memorize the first few random characters)
- The user id `{idp}-{sub}` is not super quick to get and put in a parameter: copy `idp`, paste, type `-`, copy `sub`, paste
- Another problem with email is that the user may be using another email address to write from
- We could provide instructions in the GUI to include the resource id when asking for help
  - We could show the user id and ask them to include that too
    - The `{idp}-{sub}` string gets kinda ugly, we could base64-encode it or something if that's better. That would also be helpful if the need to use it in filenames should come up again.
- Mink only knows the owner of a resource, not other users in case it has been shared through SB Auth
- I think the only way to see exactly what the user sees would be to get the user's JWT. I guess we could change SB Auth so that superusers can output any user's JWT, but that feels like a bad thing?

I feel like it could be helpful to add a filter param on email, and add a text field for that in the Mink admin GUI. And provide instructions on how to effectively ask for help (= include the resource id).

**Comment by anne17** [comment]
Thanks for your input! ⭐ I think a agree with all your points.

So I think the plan should be as follows (correct me if I misinterpreted something):

- make it possible to filter resources (owned by a user) via email (backend+frontend)
- add instructions on how to ask for help (providing the email used for login and the resource ID) (frontend, maybe we could even have a button that puts some text in the user's clipboard?)
- replace the current `{idp}-{sub}` user ID with something better. Not needed for this feature if we go with email, but something I would like to have for other purposes (backend)

I think we can live with not being able to see exactly what a user sees (in case of resource sharing). If a user who is not owner of the resource wants to ask for help, maybe this should be included in the instructions/the help button thing. For admins it's always possible to lookup the owner via resource ID. I agree that having SB Auth output other user's JWT does not seem right.

**Comment by anne17** [comment]
Created issue #154 to take care of the ugly user ID issue. So the remaining TODO of this issue is:

- [ ] make it possible to filter resources (owned by a user) via the users email address
